### PR TITLE
Add dependency-check-maven plugin to detect CPEs and CVEs

### PR DESCRIPTION
**Pull request by wanglinsong**
This PR is the first step to resolve issue #19030. The goal is to expose the known vulnerabilities in current code base.

Integrate the dependency-check-maven plugin into the build process.
All detected are set to be ignored, for now, to not block the current CI processes.

One example report of detected CPEs(Common Platform Enumeration) and CVEs (Common Vulnerabilities and Exposures) can be viewed here: https://ci.ahana.dev/job/prestodb/job/presto/job/PR-19047/17/artifact/

The next steps are to gradually update the code/dependencies to fix CPEs and CVEs, or to suppress CPEs and CVEs if they are considered as false positives.
One possible approach is to lower `failBuildOnCVSS` score in the plugin configuration to expose the detected CVSS.

Plugin reference: http://jeremylong.github.io/DependencyCheck/.

### Background information.
CVE: Common Vulnerabilities and Exposures. The mission of the CVE® Program is to identify, define, and catalog publicly disclosed cybersecurity vulnerabilities. (from https://www.cve.org/About/Overview).

CPE: Common Platform Enumeration. A standardized method of describing and identifying classes of applications, operating systems, and hardware devices present among an enterprise's computing assets. (from https://csrc.nist.gov/projects/security-content-automation-protocol/specifications/cpe#:~:text=Common%20Platform%20Enumeration%20(CPE)%20is,among%20an%20enterprise's%20computing%20assets.)

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by yzhang1991 — APPROVED** [review]

File: jenkins/agent-maven.yaml [hunk 1]
```diff
@@ -13,14 +13,14 @@ spec:
...
       env:
       - name: MAVEN_OPTS
-        value: "-Xmx6000m -Xms6000m"
+        value: "-Xmx8000m -Xms8000m"
       resources:
         requests:
-          memory: "8Gi"
-          cpu: "4000m"
+          memory: "10Gi"
+          cpu: "2000m"
```
Comment (yingsu00) [inline jenkins/agent-maven.yaml:20]: Why is cpu reduced?

**Review by yingsu00 — COMMENTED** [review]
@wanglinsong Could you please explain a bit what CPE and CVE is, why the problem happened, and why this change help resolving that issue in the commit message and PR message?
Also, in Presto we usually don't put the `[issue-19030]` in the commit and PR titles, but put it in the PR/commit body. Write it like Issue #19030 then the link would be automatically added.

File: jenkins/agent-maven.yaml (hunk 1 above)
Comment (wanglinsong) [reply jenkins/agent-maven.yaml:20]: The goal of this maven plugin integration is to expose components with known vulnerabilities. Check online document here http://jeremylong.github.io/DependencyCheck/.

**Review by yingsu00 — APPROVED** [review]
The PR and commit title needs to start with Upper case letter,  I changed it to “Add dependency-check-maven plugin to detect CPEs and CVEs”.
Also added the definition of CPE and CVE right after them in the PR message.

**Comment by vermapratyush** [comment]
@wanglinsong Hi,
This PR is causing build failures on my mac.

```
[WARNING] Unable to update 1 or more Cached Web DataSource, using local data instead. Results may not include recent vulnerabilities.
[ERROR] Unable to continue dependency-check analysis.
....
...
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:8.0.2:check (default) on project presto-root: Fatal exception(s) analyzing presto-root: One or more exceptions occurred during analysis:
[ERROR] 	UpdateException: org.owasp.dependencycheck.utils.DownloadFailedException: Error downloading file https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json; unable to connect.
[ERROR] 		caused by DownloadFailedException: Error downloading file https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json; unable to connect.
[ERROR] 		caused by UnknownHostException: fwdproxy
[ERROR] 	NoDataException: No documents exist
```

Any suggestions?

**Comment by MnO2** [comment]
332716800_1271554710373567_2857786375723329973_n

Hi @wanglinsong @yingsu00 @yzhang1991 , May we revert this commit for now given that we noticed that it has failed the open source PR CI tests. Also internally at Meta, we have many users reported that this cause build failure.

**Comment by MnO2** [comment]
I've reverted this for now @wanglinsong , @yzhang1991 @yingsu00 since it is blocking many Meta presto team members' PRs.

**Pull request merged**